crates/autopilot/src/domain/settlement/solution/mod.rs

```diff
@@ -44,20 +45,54 @@ impl Solution {
         )?)
     }
 
-    pub fn native_surplus(&self, auction: &super::Auction) -> Result<eth::Ether, trade::Error> {
+    /// Total surplus for all trades in the solution.
+    ///
+    /// Always returns a value, even if some trades have incomplete surplus
+    /// calculation.
+    pub fn native_surplus(&self, auction: &super::Auction) -> eth::Ether {
         self.trades
             .iter()
-            .map(|trade| trade.native_surplus(auction))
+            .map(|trade| {
+                trade.native_surplus(auction).unwrap_or_else(|err| {
```

Comment:
Won't this soft fail/ignore of the error make for some cases the score / surplus calculation not precise?

Reply:
Yes, that is the expected behaviour as Observation object always needs to return valid value (see existing behaviour in OnSettlementEventUpdater).

Also, I added the comment above the function signature ☝️

Reply:
Great! thank you! just wanted to really double check :ok_hand: